code/model/EditableCustomRule.php

```diff
@@ -147,4 +147,94 @@ public function canUnpublish($member = null)
     {
         return $this->canDelete($member);
     }
+
+    /**
+     * Substitutes configured rule logic with it's JS equivalents and returns them as array elements
+     * @return array
+     */
+    public function buildExpression()
+    {
+        /** @var EditableFormField $formFieldWatch */
+        $formFieldWatch = $this->ConditionField();
+        //Encapsulated the action to the object
+        $action = $formFieldWatch->getJsEventHandler();
+
+        // is this field a special option field
+        $checkboxField = $formFieldWatch->isCheckBoxField();
+        $radioField = $formFieldWatch->isRadioField();
+
+        $target = sprintf('$("%s")', $formFieldWatch->getSelectorFieldOnly());
+
+        // and what should we evaluate
+        switch ($this->ConditionOption) {
+            case 'IsNotBlank':
+                $expression = ($checkboxField || $radioField) ? "{$target}.is(\":checked\")" : "{$target}.val() != ''";
+
+                break;
+            case 'IsBlank':
+                $expression = ($checkboxField || $radioField) ? "!({$target}.is(':checked'))" : "{$target}.val() == ''";
+
+                break;
+            case 'HasValue':
+                if ($checkboxField) {
+                    $expression = "{$target}.prop('checked')";
+                } elseif ($radioField) {
+                    // We cannot simply get the value of the radio group, we need to find the checked option first.
+                    $expression = sprintf('%s.closest(".field, .control-group").find("input:checked").val()=="%s"',
+                        $target, $this->FieldValue);
+                } else {
+                    $expression = sprintf('%s.val() == "%s"', $target, $this->FieldValue);
+                }
+
+                break;
+            case 'ValueLessThan':
+                $expression = sprintf('%s.val() < parseFloat("%s")', $target, $this->FieldValue);
+
+                break;
+            case 'ValueLessThanEqual':
+                $expression = sprintf('%s.val() <= parseFloat("%s")', $target, $this->FieldValue);
+
+                break;
+            case 'ValueGreaterThan':
+                $expression = sprintf('%s.val() > parseFloat("%s")', $target, $this->FieldValue);
+
+                break;
+            case 'ValueGreaterThanEqual':
+                $expression = sprintf('%s.val() >= parseFloat("%s")', $target, $this->FieldValue);
+
+                break;
+            default: // ==HasNotValue
```

Comment:
Maybe instead of assuming, we should have `case 'ValueNot'` and error in `default:` if not in the valid range. Also, `HasNotValue` isn't a valid option, so that comment is wrong anyway.